superset-frontend/spec/javascripts/views/CRUD/data/database/DatabaseModal_spec.jsx

```diff
@@ -19,71 +19,243 @@
+      expect(wrapper.find('input[name="sqlalchemy_uri"]')).toExist();
+    });
   });
 
-  it('renders a Modal', () => {
-    expect(wrapper.find(Modal)).toExist();
-  });
+  describe('RTL', () => {
+    describe('initial load', () => {
+      it('hides the forms from the db when not selected', () => {
+        render(
+          <ThemeProvider theme={supersetTheme}>
+            <Provider store={store}>
+              <DatabaseModal
+                show
+                database={{
+                  expose_in_sqllab: false,
+                  allow_ctas: false,
+                  allow_cvas: false,
+                }}
+              />
+            </Provider>
+          </ThemeProvider>,
+        );
+        // Select SQL Lab settings tab
+        const sqlLabSettingsTab = screen.getByRole('tab', {
+          name: /sql lab settings/i,
+        });
+        userEvent.click(sqlLabSettingsTab);
 
-  it('renders "Add database" header when no database is included', () => {
-    expect(wrapper.find('h4').text()).toEqual('Add database');
-  });
+        const exposeInSqlLab = screen.getByText('Expose in SQL Lab');
+        const exposeChoicesForm = exposeInSqlLab.parentElement.nextSibling;
+        const schemaField = screen.getByText('CTAS & CVAS SCHEMA')
+          .parentElement;
+        expect(exposeChoicesForm).not.toHaveClass('open');
+        expect(schemaField).not.toHaveClass('open');
+      });
+    });
+    it('renders all settings when "Expose in SQL Lab" is checked', () => {
+      render(
+        <ThemeProvider theme={supersetTheme}>
+          <Provider store={store}>
+            <DatabaseModal {...dbProps} />
+          </Provider>
+        </ThemeProvider>,
+      );
 
-  it('renders "Edit database" header when database prop is included', () => {
-    const editWrapper = mount(<DatabaseModal store={store} {...dbProps} />);
-    waitForComponentToPaint(editWrapper);
-    expect(editWrapper.find('h4').text()).toEqual('Edit database');
-  });
+      // Select SQL Lab settings tab
+      const sqlLabSettingsTab = screen.getByRole('tab', {
+        name: /sql lab settings/i,
+      });
 
-  it('renders a Tabs menu', () => {
-    expect(wrapper.find(Tabs)).toExist();
-  });
+      userEvent.click(sqlLabSettingsTab);
+      // Grab all SQL Lab Settings by their labels
+      // const exposeInSqlLab = screen.getByText('Expose in SQL Lab');
+      const exposeInSqlLab = screen.getByRole('checkbox', {
+        name: /expose in sql lab/i,
+      });
 
-  it('renders five TabPanes', () => {
-    expect(wrapper.find(Tabs.TabPane)).toExist();
-    expect(wrapper.find(Tabs.TabPane)).toHaveLength(5);
-  });
+      // While 'Expose in SQL Lab' is checked, all settings should display
+      expect(exposeInSqlLab).not.toBeChecked();
+
+      // When clicked, "Expose in SQL Lab" becomes unchecked
+      userEvent.click(exposeInSqlLab);
+
+      // While unchecked, only "Expose in SQL Lab" should display
```

Comment:
Are the comments still aligned with the code? This says unchecked, while the assertion below is checked.